__tests__/app.test.js

```diff
@@ -164,10 +164,88 @@ describe("GET /api/articles/:article_id/comments", () => {
   })
   test("400: Article_id is not a number",()=>{
     return request(app)
-    .get("/api/articles/its-over-nine-thousand/comments")
+    .get("/api/articles/not-a-number/comments")
     .expect(400)
     .then(({body:{message}})=>{
       expect(message).toBe("Bad request")
     })
   })
 });
+
+describe("POST /api/articles/:article_id/comments",()=>{
+  test("201: Post comment onto an article and respond with the posted comment",()=>{
+    return request(app)
+    .post("/api/articles/1/comments")
+    .send({
+      username:"lurker",
+      body:"test comment body"
+    })
+    .expect(201)
+    .then(({body:{comment}})=>{
+      expect(comment).toEqual({
+        article_id:1,
+        votes:0,
+        created_at:expect.any(String),
+        comment_id:expect.any(Number),
+        author:"lurker",
+        body:"test comment body"
+      })
+    })
+  });
+  test("404: Article_id does not exist",()=>{
+    return request(app)
+    .post("/api/articles/9001/comments")
+    .send({
+      username:"lurker",
+      body:"test comment body"
+    })
+    .expect(404)
+    .then(({body:{message}})=>{
+      expect(message).toBe("Article Id does not exist")
+    })
+  })
+  test("400: Article_id is not a number",()=>{
+    return request(app)
+    .post("/api/articles/not-a-number/comments")
+    .send({
+      username:"lurker",
+      body:"test comment body"
+    })
+    .expect(400)
+    .then(({body:{message}})=>{
+      expect(message).toBe("Bad request")
+    })
+  })
+  test("400: Invalid user",()=>{
+    return request(app)
+    .post("/api/articles/1/comments")
+    .send({
+      username:"lurk",
+      body:"test comment body"
+    })
+    .expect(401)
```

Comment:
This code doesn't match what's in your test description. If the username doesn't exist, this would actually be a 404. PSQL should throw an error for you that you can catch 👍